src/Microsoft.DotNet.Interactive/Connection/KernelCommandAndEventTextStreamReceiver.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.Server;
+
+namespace Microsoft.DotNet.Interactive.Connection
+{
+    public class CommandOrEvent
+    {
+        public KernelCommand Command { get; }
+        public KernelEvent Event { get; }
+
+        public CommandOrEvent(KernelCommand kernelCommand, KernelEvent kernelEvent)
```

Comment:
This would be clearer if there were two separate constructors. Make the invalid call fail at compile time instead of runtime.

Reply:
done